src/parser/core/modules/Weaving.tsx

```diff
@@ -0,0 +1,241 @@
+import {t} from '@lingui/macro'
+import {Trans, Plural} from '@lingui/react'
+import NormalisedMessage from 'components/ui/NormalisedMessage'
+import Rotation from 'components/ui/Rotation'
+import {Action} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import CastTime from 'parser/core/modules/CastTime'
+import {Data} from 'parser/core/modules/Data'
+import {Invulnerability} from 'parser/core/modules/Invulnerability'
+import Suggestions, {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import React, {ReactNode} from 'react'
+import {Accordion} from 'semantic-ui-react'
+import {matchClosestLower} from 'utilities'
+import {BASE_GCD} from 'utilities/speedStatMapper'
+
+const CAST_TIME_MAX_WEAVES = {
+	0: 2,
+	1000: 1,
+	2500: 0,
+}
+const REDUCE_MAX_WEAVES_RECAST_BELOW = 2000
+const DEFAULT_MAX_WEAVES = 2
+
+const WEAVING_SEVERITY = {
+	1: SEVERITY.MEDIUM,
+	5: SEVERITY.MAJOR,
+}
+
+export interface Weave {
+	leadingGcdEvent?: Events['action'],
+	trailingGcdEvent: Events['action'],
+	gcdTimeDiff: number,
+	weaves: Array<Events['action']>,
+}
+
+export class Weaving extends Analyser {
+	static override handle = 'weaving'
+
+	@dependency protected castTime!: CastTime
+	@dependency private data!: Data
+	@dependency private invulnerability!: Invulnerability
+	@dependency private suggestions!: Suggestions
+
+	static override title = t('core.weaving.title')`Weaving Issues`
+
+	protected suggestionIcon: string = 'https://xivapi.com/i/001000/001785.png' // WVR Focused Synth
+
+	protected moduleLink = <a
+		style={{cursor: 'pointer'}}
+		onClick={() => this.parser.scrollTo(Weaving.handle)}><NormalisedMessage message={Weaving.title}/>
+	</a>
+	protected suggestionContent: ReactNode = <Trans id="core.weaving.content">
```

Comment:
I'd suggest a comment much like buffwindow that lets people know they can override this – this is me suggesting something non-core dev-facing though.